### PR TITLE
Passing data to send to websocket

### PR DESCRIPTION
**Pull request by finga**
When using `WebSocketTask::send()` and `WebSocketTask::send_binary()` it feels rather strange to pass the data as a `Result` to the function. Would it not make more sense to pass `String` and `Vec<u8>` respectively?

Please let me know what you think about that :)

### CODE REVIEW

**Comment by darioalessandro** [comment]
I agree, I took this pattern straight from the yew repo. 

Do we need to keep those Json macros? https://github.com/security-union/yew-websocket/blob/master/src/macros.rs

Is there any dead code after this?

**Comment by finga** [comment]
Glad to hear, I will take a look if those macros are still needed and if there will be any dead code. I just wanted to wait for your point of view on this before digging any deeper. Further, I will adapt the example to reflect those changes as well. I will let you know here when I think I am done with everything. Hope that is fine for you.

One last thing, since this is a breaking change, when releasing this, I think an increase in the major version number is needed, but I am not entirely sure as I am not too familiar with semver.

**Comment by darioalessandro** [comment]
sounds good, lets increase the major version, no big deal, it is early enough

**Comment by finga** [comment]
I pushed what I did with the example and the macros. I am not fully happy with it, as I wanted to get rid of the whole `Json` derive macro code, but I could not find a better solution for that. What do you think about it?

**Comment by finga** [comment]
Just wanted to ask what you think about my suggestions. If you are fine with them feel free to merge :)

**Comment by darioalessandro** [comment]
Roger that! Will review today

**Comment by darioalessandro** [comment]
@finga looking good!!

**Comment by finga** [comment]
> @finga looking good!!

Awesome, thx!!

**Comment by finga** [comment]
I took a look at the dependencies and took the liberty to update them. If you do not like how I did it just let me know and I can remove 3487229. Otherwise I think this should be ready to be merged.

**Comment by darioalessandro** [comment]
Sounds good, I'll test locally and add a couple of github actions, then I'll go ahead and merge it

**Comment by finga** [comment]
I do not want to be annoying, just wondering when I can count on having this released?

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 .DS_Store
-target
\ No newline at end of file
+target
+.gitignore
```
Comment (darioalessandro) [inline .gitignore:3]: adding gitignore to gitignore? I'll remove this entry

**Review by darioalessandro — APPROVED** [review]
lgtm

**Comment by darioalessandro** [comment]
@finga It is out!!! https://crates.io/crates/yew-websocket 

Sorry for taking this long!

**Comment by darioalessandro** [comment]
It is out @finga !!

**Pull request merged**